### ISSUE TITLE
Visibility refactor in #868 broke compilation under MinGW GCC

### ISSUE THREAD

**Issue by amyspark** [AcademySoftwareFoundation/openexr#1234]
Hi,

In #868 the visibility system was completely overhauled. Unfortunately, the assumption that `default`-defined destructors can be manually exported breaks building with MinGW GCC.

```
In file included from E:/krita-win/b_deps_msys/ext_openexr/ext_openexr-prefix/src/ext_openexr/OpenEXR/IlmImf/ImfKeyCodeAttribute.h:47,
                 from E:/krita-win/b_deps_msys/ext_openexr/ext_openexr-prefix/src/ext_openexr/OpenEXR/IlmImf/ImfStandardAttributes.h:63,
                 from E:\krita-win\b_deps_msys\ext_openexr\ext_openexr-prefix\src\ext_openexr\OpenEXR\exrenvmap\readInputImage.cpp:46:
E:/krita-win/b_deps_msys/ext_openexr/ext_openexr-prefix/src/ext_openexr/OpenEXR/IlmImf/ImfKeyCode.h:124:5: error: function 'Imf_2_5::KeyCode::~KeyCode()' definition is marked dllimport
  124 |     ~KeyCode() = default;
      |   
```

OpenEXR version: `2.5.5`
GCC version: `g++.exe (Rev9, Built by MSYS2 project) 11.2.0`

Probably cc @cary-ilm because it seems a validation of #358.

**Comment by amyspark** [comment]
This bit of https://github.com/AcademySoftwareFoundation/openexr/commit/613120445f10bf6cb960ff58f18ce2366bf3455d#diff-fd8c86a92e7aaa248ad4308e57dfcdff01e15f5f79cfb3f0f47ec84ad6052ce8L93 needs to be cherry-picked into 2.5. I'll PR it.

**Comment by amyspark** [comment]
Fixed by #1235
